### PR TITLE
Backport Python 3.11 and Django 4.2 in v2

### PR DESCRIPTION
**Pull request by wongcht**
v2.16.0 supports Django 4.1, actually it also passes the tests in Python 3.11 and Django 4.2.

Suggest changing the latest supported version to cover Django 4.2 LTS, while currently 4.1 EOL in 3 months (2023-12-1).

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -38,11 +42,12 @@ deps =
     django32: Django>=3.2,<4.0
     django40: Django>=4.0,<4.1
     django41: Django>=4.1.3,<4.2
+    django42: Django>=4.2,<5.0
```
Comment (kiendang) [inline tox.ini:45]:
```suggestion
    django42: Django>=4.2,<4.3
```

File: tox.ini [hunk 2]
```diff
@@ -3,7 +3,9 @@ envlist =
     py{37,38,39}-django22,
     py{37,38,39}-django{30,31},
     py{37,38,39,310}-django32,
-    py{38,39,310}-django{40,41,master},
+    py{38,39,310}-django40,
+    py{38,39,310,311}-django{41,42},
+    py{310,311}-djangomaster,
```
Comment (kiendang) [inline tox.ini:8]:
```suggestion
    py{310,311}-main,
```

File: tox.ini [hunk 3]
```diff
@@ -21,6 +24,7 @@ DJANGO =
     3.2: django32
     4.0: django40
     4.1: django41
+    4.2: django42
     master: djangomaster
```
Comment (kiendang) [inline tox.ini:28]:
```suggestion
    main: djangomain
```

File: tox.ini [hunk 4]
```diff
@@ -38,11 +42,12 @@ deps =
     django32: Django>=3.2,<4.0
     django40: Django>=4.0,<4.1
     django41: Django>=4.1.3,<4.2
+    django42: Django>=4.2,<5.0
     djangomaster: https://github.com/django/django/archive/master.zip
```
Comment (kiendang) [inline tox.ini:46]:
```suggestion
    djangomain: https://github.com/django/django/archive/main.zip
```

**Review by kiendang — COMMENTED** [review]
Thanks @wongcht! Just a few suggestions.

File: tox.ini [hunk 5]
```diff
@@ -3,7 +3,9 @@ envlist =
     py{37,38,39}-django22,
     py{37,38,39}-django{30,31},
     py{37,38,39,310}-django32,
-    py{38,39,310}-django{40,41,master},
+    py{38,39,310}-django40,
+    py{38,39,310,311}-django{41,42},
+    py{310,311}-main,
```
Comment (wongcht) [inline tox.ini:8]:
```suggestion
    py{310,311}-djangomain,
```
@kiendang should it actually be `djangomain`?

Comment (kiendang) [reply tox.ini:8]: Yup you're right.

**Review by kiendang — APPROVED** [review]

**Review by firaskafri — APPROVED** [review]

**Comment by cpd67** [comment]
@firaskafri @kiendang Hey guys, I apologize for commenting on a closed MR, but will this be part of a v2 release soon? We're in the middle of attempting to upgrade to Django 4.2, but currently cannot because the current graphene-django v2 release doesn't support Django 4.2.

**Comment by kiendang** [comment]
@cpd67 This PR only added tests for Django 4.2 to CI and made no changes to the code. You could already use the current `graphene-django` v2 release with Django 4.2. I suggest you try that and see if there are problems.

> We've tried using graphene-django v3, but ran into issues because some of our resolvers are async and it looks like graphene-django v3 doesn't support async resolvers yet.

We currently have a PR open for adding async to v3 #1394 which we are considering releasing as an experimental feature soon (see [this discussion](https://github.com/graphql-python/graphene-django/discussions/1449)). Are you interested in trying that out? Would help us a lot. You can try the PR right now with 

```sh
pip install git+https://github.com/jaw9c/graphene-django.git@support-async
```

**Comment by cpd67** [comment]
@kiendang Thanks for the reply! And apologies for misunderstanding the purpose of this PR :sweat_smile:.

I was aware of that PR and attempted to contribute a bit through this [PR](https://github.com/jaw9c/graphene-django/pull/1). I can certainly test it out on a local project of mine and offer feedback.

**Comment by kiendang** [comment]
@cpd67 I see. Yup I'm aware of your PR, just didn't recognize it was you haha.

> I can certainly test it out on a local project of mine and offer feedback.

Thank you so much!

**Pull request merged**